### PR TITLE
Fixed various code warnings

### PR DESCRIPTION
**Pull request by tlusk**
Most of these are harmless little things like null checks on things that can never be null and docstrings missing, but there were a couple bugs I found while cleaning this up.

### CODE REVIEW

File: src/main/java/org/jenkinsci/plugins/ghprb/GhprbRepository.java [hunk 1]
```diff
@@ -290,8 +290,8 @@ public boolean createHook() {
             String secret = getSecret();
             config.put("url", new URL(getHookUrl()).toExternalForm());
             config.put("insecure_ssl", "1");
-            if (secret != "") {
-             config.put("secret",secret);
+            if (secret != null && !secret.equals("")) {
+             config.put("secret", secret);
```
Comment (tlusk) [inline src/main/java/org/jenkinsci/plugins/ghprb/GhprbRepository.java]: There was a real bug here, comparing a string with `!=`

File: src/main/java/org/jenkinsci/plugins/ghprb/upstream/GhprbUpstreamStatusListener.java [hunk 2]
```diff
@@ -59,15 +60,15 @@ private boolean updateEnvironmentVars(AbstractBuild<?, ?> build, TaskListener li
...
         
         if (StringUtils.isEmpty(context)) {
             context = jobName;
         }
         
-        Boolean addTestResults = new Boolean(envVars.get("ghprbStartedStatus"));
+        Boolean addTestResults = Boolean.valueOf(envVars.get("ghprbStartedStatus"));
 
-        statusUpdater = new GhprbSimpleStatus(envVars.get("ghprbCommitStatusContext"), envVars.get("ghprbStatusUrl"), envVars.get("ghprbTriggeredStatus"), envVars.get("ghprbStartedStatus"), addTestResults, statusMessages);
+        statusUpdater = new GhprbSimpleStatus(context, envVars.get("ghprbStatusUrl"), envVars.get("ghprbTriggeredStatus"), envVars.get("ghprbStartedStatus"), addTestResults, statusMessages);
```
Comment (tlusk) [inline src/main/java/org/jenkinsci/plugins/ghprb/upstream/GhprbUpstreamStatusListener.java]: This was a bug too, the context value above was being set to the job name if it was empty, but then not being used.

File: src/main/java/org/jenkinsci/plugins/ghprb/GhprbPullRequestMerge.java [hunk 3]
```diff
@@ -37,11 +37,11 @@
 
     private transient PrintStream logger;
 
-    private final Boolean onlyAdminsMerge;
-    private final Boolean disallowOwnCode;
+    private final boolean onlyAdminsMerge;
```
Comment (DavidTanner) [inline src/main/java/org/jenkinsci/plugins/ghprb/GhprbPullRequestMerge.java]: In my experience it is preferable to us the object here instead of a primitive incase it is null when the get method is called.  This is possible because Jenkins doesn't use the constructor to recreate objects when it restarts.

File: src/main/java/org/jenkinsci/plugins/ghprb/GhprbPullRequestMerge.java [hunk 4]
```diff
@@ -151,10 +150,10 @@ public boolean perform(AbstractBuild<?, ?> build, Launcher launcher, final Build
             }
         }
 
-        Boolean isMergeable = cause.isMerged();
+        boolean isMergeable = cause.isMerged();
```
Comment (DavidTanner) [inline src/main/java/org/jenkinsci/plugins/ghprb/GhprbPullRequestMerge.java]: Again this can become an NPE if isMerged doesn't return a boolean.

File: src/main/java/org/jenkinsci/plugins/ghprb/GhprbRepository.java (hunk 1 above)
Comment (DavidTanner) [inline src/main/java/org/jenkinsci/plugins/ghprb/GhprbRepository.java]: The preference is to use StringUtils.isEmpty()

File: src/main/java/org/jenkinsci/plugins/ghprb/GhprbRootAction.java [hunk 5]
```diff
@@ -152,13 +152,11 @@ public void doIndex(StaplerRequest req, StaplerResponse resp) {
     }
 
     private PullRequest getPullRequest(String payload, GitHub gh) throws IOException {
-        PullRequest pr = gh.parseEventPayload(new StringReader(payload), PullRequest.class);
-        return pr;
+        return gh.parseEventPayload(new StringReader(payload), PullRequest.class);
```
Comment (DavidTanner) [inline src/main/java/org/jenkinsci/plugins/ghprb/GhprbRootAction.java]: I prefer to have the variable on the line before for debugging.

File: src/main/java/org/jenkinsci/plugins/ghprb/GhprbTrigger.java [hunk 6]
```diff
@@ -372,11 +370,10 @@ public String getGitHubAuthId() {
     
     public GhprbGitHubAuth getGitHubApiAuth() {
         if (gitHubAuthId == null) {
-            for (GhprbGitHubAuth auth: getDescriptor().getGithubAuth()){
-                gitHubAuthId = auth.getId();
-                getDescriptor().save();
-                return auth;
-            }
+            GhprbGitHubAuth auth = getDescriptor().getGithubAuth().get(0);
```
Comment (DavidTanner) [inline src/main/java/org/jenkinsci/plugins/ghprb/GhprbTrigger.java]: This assumes that there will always be at least one entry, which it is possible there may not be.

File: src/main/java/org/jenkinsci/plugins/ghprb/extensions/GhprbBuildStep.java [hunk 7]
```diff
@@ -6,7 +6,7 @@
 import hudson.model.Action;
 
 public interface GhprbBuildStep extends Action {
-    public static String buildStep = "GhprbBuildStep";
+    String buildStep = "GhprbBuildStep";
```
Comment (DavidTanner) [inline src/main/java/org/jenkinsci/plugins/ghprb/extensions/GhprbBuildStep.java]: Please leave as public methods.  There is no reason to mess with these.

File: src/main/java/org/jenkinsci/plugins/ghprb/extensions/comments/GhprbPublishJenkinsUrl.java [hunk 8]
```diff
@@ -33,13 +33,9 @@ public String getPublishedURL() {
...
 
     public String postBuildComment(AbstractBuild<?, ?> build, TaskListener listener) {
-        StringBuilder msg = new StringBuilder();
-
-        msg.append("\nRefer to this link for build results (access rights to CI server needed): \n");
-        msg.append(generateCustomizedMessage(build));
-        msg.append("\n");
-        
-        return msg.toString();
+        return "\nRefer to this link for build results (access rights to CI server needed): \n" +
```
Comment (DavidTanner) [inline src/main/java/org/jenkinsci/plugins/ghprb/extensions/comments/GhprbPublishJenkinsUrl.java]: The StringBuilder is cleaner and more readable.  It also allows for easier expansion or editing.

File: src/main/java/org/jenkinsci/plugins/ghprb/upstream/GhprbUpstreamStatusListener.java (hunk 2 above)
Comment (DavidTanner) [inline src/main/java/org/jenkinsci/plugins/ghprb/upstream/GhprbUpstreamStatusListener.java]: It would make more sense then to remove line 63 and keep with the rest of the arguments on line 71, or vice versa

File: src/main/java/org/jenkinsci/plugins/ghprb/upstream/GhprbUpstreamStatusListener.java [hunk 9]
```diff
@@ -115,7 +116,7 @@ public void onStarted(AbstractBuild<?, ?> build, TaskListener listener) {
 
     // Sets the status to the build result when the job is done, and then calls the createCommitStatus method to send it to GitHub
     @Override
-    public void onCompleted(AbstractBuild<?, ?> build, TaskListener listener) {
+    public void onCompleted(AbstractBuild<?, ?> build, @Nonnull TaskListener listener) {
```
Comment (DavidTanner) [inline src/main/java/org/jenkinsci/plugins/ghprb/upstream/GhprbUpstreamStatusListener.java]: What does this accomplish?

File: src/test/java/org/jenkinsci/plugins/ghprb/manager/impl/downstreambuilds/BuildFlowBuildManagerTest.java [hunk 10]
```diff
@@ -43,15 +43,13 @@ public void setUp() throws Exception {
         jenkinsRule.createFreeStyleProject("downstreamProject2");
         jenkinsRule.createFreeStyleProject("downstreamProject3");
 
-        StringBuilder dsl = new StringBuilder();
+        String dsl = "parallel (" +
```
Comment (DavidTanner) [inline src/test/java/org/jenkinsci/plugins/ghprb/manager/impl/downstreambuilds/BuildFlowBuildManagerTest.java]: Again, the StringBuilder is faster and cleaner to read.

File: src/main/java/org/jenkinsci/plugins/ghprb/extensions/GhprbBuildStep.java (hunk 7 above)
Comment (tlusk) [inline src/main/java/org/jenkinsci/plugins/ghprb/extensions/GhprbBuildStep.java]: All methods in an interface are public, it's redundant as it can not be anything but public.

File: src/main/java/org/jenkinsci/plugins/ghprb/GhprbTrigger.java (hunk 6 above)
Comment (tlusk) [inline src/main/java/org/jenkinsci/plugins/ghprb/GhprbTrigger.java]: That would be true, except in the current behavior if there aren't any it will fall through this and call `getDescriptor().getGitHubAuth(gitHubAuthId);` below with a null value.  And if you look in that method it returns `getDescriptor().getGithubAuth().get(0);` if a null is passed.  So basically that ends up getting called no matter what.  I'll take another look here and see if there's a better way to do it but if that's a concern there's most likely a bug here already.

File: src/main/java/org/jenkinsci/plugins/ghprb/GhprbRepository.java (hunk 1 above)
Comment (tlusk) [inline src/main/java/org/jenkinsci/plugins/ghprb/GhprbRepository.java]: Noted, will change.

File: src/main/java/org/jenkinsci/plugins/ghprb/GhprbPullRequestMerge.java (hunk 4 above)
Comment (tlusk) [inline src/main/java/org/jenkinsci/plugins/ghprb/GhprbPullRequestMerge.java]: `isMerged()` returns a `boolean`, not a `Boolean`.  So there's no possibility of NPE.

File: src/main/java/org/jenkinsci/plugins/ghprb/upstream/GhprbUpstreamStatusListener.java (hunk 9 above)
Comment (tlusk) [inline src/main/java/org/jenkinsci/plugins/ghprb/upstream/GhprbUpstreamStatusListener.java]: The method you're overriding has it defined.  It's better to copy the declaration of the parent method exactly.

File: src/main/java/org/jenkinsci/plugins/ghprb/extensions/comments/GhprbPublishJenkinsUrl.java (hunk 8 above)
Comment (tlusk) [inline src/main/java/org/jenkinsci/plugins/ghprb/extensions/comments/GhprbPublishJenkinsUrl.java]: For something this small I'm not sure I agree, but it's just splitting hairs so I'll change it back.

File: src/main/java/org/jenkinsci/plugins/ghprb/GhprbRootAction.java (hunk 5 above)
Comment (tlusk) [inline src/main/java/org/jenkinsci/plugins/ghprb/GhprbRootAction.java]: I was wondering what the reason behind this was.  I'll change it back and suppress the warnings for these.

File: src/main/java/org/jenkinsci/plugins/ghprb/upstream/GhprbUpstreamStatusListener.java (hunk 2 above)
Comment (tlusk) [inline src/main/java/org/jenkinsci/plugins/ghprb/upstream/GhprbUpstreamStatusListener.java]: Not sure I understand this suggestion.  How can you remove 63 -> 67 and still pass the job name as the context if the context is empty?  Without some really ugly inline if statement.

`new GhprbSimpleStatus(StringUtils.isEmpty(envVars.get("ghprbCommitStatusContext")) ? envVars.get("ghprbCommitStatusContext") : jobName, envVars.get("ghprbStatusUrl"), etc...`

**Comment by tlusk** [comment]
Updated based on feedback.  I'll squash these together when I get a :+1: before merging.

File: src/main/java/org/jenkinsci/plugins/ghprb/GhprbTrigger.java [hunk 11]
```diff
@@ -371,14 +364,12 @@ public String getGitHubAuthId() {
...
-                getDescriptor().save();
-                return auth;
-            }
+        GhprbGitHubAuth auth = getDescriptor().getGitHubAuth(gitHubAuthId);
+        if (gitHubAuthId == null && auth != null) {
+            gitHubAuthId = auth.getId();
+            getDescriptor().save();
         }
-        return getDescriptor().getGitHubAuth(gitHubAuthId);
+        return auth;
```
Comment (tlusk) [inline src/main/java/org/jenkinsci/plugins/ghprb/GhprbTrigger.java]: The changes to both this 'getGitHubApiAuth()` and `getGitHubAuth(String gitHubAuthId)` now prevent the possibility of an exception being thrown if the `List<GhprbGitHubAuth>` is empty.  (It would have thrown an exception before).

File: src/main/java/org/jenkinsci/plugins/ghprb/GhprbRootAction.java [hunk 12]
```diff
@@ -152,11 +152,13 @@ public void doIndex(StaplerRequest req, StaplerResponse resp) {
     }
 
     private PullRequest getPullRequest(String payload, GitHub gh) throws IOException {
+        @SuppressWarnings("UnnecessaryLocalVariable")
         PullRequest pr = gh.parseEventPayload(new StringReader(payload), PullRequest.class);
         return pr;
     }
 
     private IssueComment getIssueComment(String payload, GitHub gh) throws IOException {
+        @SuppressWarnings("UnnecessaryLocalVariable")
```
Comment (tlusk) [inline src/main/java/org/jenkinsci/plugins/ghprb/GhprbRootAction.java]: Restored your local variables for debugging and suppressed the warnings so it's known it's intentional.

File: src/main/java/org/jenkinsci/plugins/ghprb/GhprbPullRequestMerge.java (hunk 3 above)
Comment (tlusk) [inline src/main/java/org/jenkinsci/plugins/ghprb/GhprbPullRequestMerge.java]: Using `boolean` prevents it from the possibility of being null when the get methods are called.  And even if Jenkins is assigning these via reflection upon reboot, I can't imagine their code would be applying explicit nulls to primitives.  I'll change it back if you want but it feels like using the `Boolean` object is just asking for more trouble and requiring explicit value checks.

File: src/main/java/org/jenkinsci/plugins/ghprb/upstream/GhprbUpstreamStatusListener.java (hunk 9 above)
Comment (tlusk) [inline src/main/java/org/jenkinsci/plugins/ghprb/upstream/GhprbUpstreamStatusListener.java]: In more detail what is happening is that the parent method from jenkins-core is defined with the `@Nonnull` annotation which would cause a compile warning if a null parameter was sent to this method, because the jenkins team wants it known it's not safe to call `onCompleted()` with a *null* `TaskListener`.

Before this change the `onCompleted` override function in the plugin was stripping off the annotation saying that it was OK to pass it a *null* `TaskListener`.  It's generally not good to strip annotations when inheriting unless there's good reason to, and looking into the implementation of the plugin passing a *null* `TaskListener` to the `onCompleted` function **will ** result in an NPE.

TLDR; It's a safety check to prevent passing of nulls to functions that don't handle them.
